dla_cnn/data_model/DesiMock.py

```diff
@@ -0,0 +1,88 @@
+    :attribute split_point_rz, the length of the flux_b and flux_r, the split point of  r channel data and z channel data
+    :attribute data_size, the point number of all data points of wavelength and flux
+    """
+
+    def _init_(self, wavelength = None, data = {}, split_point_br = None, split_point_rz = None, data_size = None):
+        self.wavelength = wavelength
+        self.data = data
+        self.split_point_br = split_point_br
+        self.split_point_rz = split_point_rz
+        self.data_size = data_size
+
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
+        :param truth_path: truth file path
+        :param zbest_path: zbest file path
+        :return: self.wavelength,self.data(contained all information we need),self.split_point_br,self.split_point_rz,self.data_size
+        """
+        spec = fits.open(spec_path)
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        #spec[2].data ,spec[7].data and spec[12].data are the wavelength data
+        self.wavelength = np.hstack((spec[2].data.copy(), spec[7].data.copy(), spec[12].data.copy()))
+        self.data_size = len(self.wavelength)
+
+        dlas_data = truth[3].data[truth[3].data.copy()['NHI']>19.3]
+        spec_dlas = {}
+        #item[2] is the spec_id, item[3] is the dla_id, and item[0] is NHI, item[1] is z_qso
+        for item in dlas_data:
+            if item[2] not in spec_dlas:
+                spec_dlas[item[2]] = [('00'+str(item[3]-item[2]*1000),item[1],item[0])]
+            else:
+                spec_dlas[item[2]].append(('00'+str(item[3]-item[2]*1000),item[1],item[0]))
+
+        test = np.array([True if item in dlas_data['TARGETID'] else False for item in spec[1].data['TARGETID'].copy()])
+        for item in spec[1].data['TARGETID'].copy()[~test]:
+            spec_dlas[item] = []
+
+        spec_id = spec[1].data['TARGETID'].copy()
+        flux_b = spec[3].data.copy()
+        flux_r = spec[8].data.copy()
+        flux_z = spec[13].data.copy()
+        flux = np.hstack((flux_b,flux_r,flux_z))
+        ivar_b = spec[4].data.copy()
+        ivar_r = spec[9].data.copy()
+        ivar_z = spec[14].data.copy()
+        error = 1./np.sqrt(np.hstack((ivar_b,ivar_r,ivar_z)))
+        self.split_point_br = len(flux_b)
+        self.split_point_rz = len(flux_b)+len(flux_z)
+        z_qso = zbest[1].data['Z'].copy()
+        ra = spec[1].data['TARGET_RA'].copy()
+        dec = spec[1].data['TARGET_DEC'].copy()
+
+        self.data = {spec_id[i]:{'FLUX':flux[i],'ERROR': error[i], 'z_qso':z_qso[i] , 'RA': ra[i], 'DEC':dec[i], 'DLAS':spec_dlas[spec_id[i]]} for i in range(len(spec_id))}
+
+        return self.wavelength,self.data,self.split_point_br,self.split_point_rz,self.data_size
+
+    def get_sightline(self,id):

```

Comment:
Modify this method to allow the user to grab only 1 of the 3 cameras.
It can default to all 3.  e.g.

`def get_sightline(self,id,camera='all'):`

then

`if camera == 'b':`
  Load up only the wavelength and data for the blue camera.

Reply:
I have changed `get_sightline(self,id,camera='all')` to allow the user grab the camera they want.  Thanks for your advice!